nmtwizard/preprocess/tu.py

```diff
@@ -204,3 +288,42 @@ def metadata(self, metadata):
     @property
     def annotations(self):
         return self.__annotations
+
+    def replace_tokens(self,
+                       src_replace = None, # TokReplace structure
+                       tgt_replace = None, # TokReplace structure
+                       part = 0): # TODO : maybe rather send multi-part replacements ?
+
+        # Replace (delete, insert) tokens in a TU and adjust alignment without retoknization/realignment.
+
+        if src_replace:
+            # replace tokens in source and adjust alignment if any
+            src_replace = TokReplace(*src_replace)
+            self.replace_tokens_side("source", src_replace, part=part)
+
+        if tgt_replace:
+            # replace tokens in source and adjust_alignment if any
+            tgt_replace = TokReplace(*tgt_replace)
+            self.replace_tokens_side("target", tgt_replace, part=part)
+
+        # TODO
+        # Maybe provide and alignment for inserted tokens ?
+
+
+    def replace_tokens_side(self, side, replacement, part=0):
+
+        if side == "source":
+            trans_side = self.__source
+        elif side == "target":
+            trans_side = self.__target
+
+        # Initialize alignment
+        alignment = self.alignment
+
+        # Replace tokens
+        trans_side.replace_tokens(*replacement, part=part)
+
+        # Adjust alignment
+        side_idx = 0 if side == "source" else 1
```

Comment:
Can this be moved to the `if side == ...` above?

Reply:
Look ok to you now ?

Reply:
Looks ok.